### ISSUE TITLE
Can't get Grid sizing scopes to work in FuncUI

### ISSUE THREAD

**Issue by marklam** [fsprojects/Avalonia.FuncUI#443]
I've made a sample repo at https://github.com/marklam/GridSizingScopes

The sample code from https://docs.avaloniaui.net/docs/reference/controls/grid/sharedsizegroup is replicated in F# in the XamlGrids project.

An attempt to recreate this in FuncUI is in the FuncUIGrids project.

The XamlGrids version sizes the groups A, B and C (columns 1, 2 and 4) to their content. The FuncUI version sizes all the columns as if they were of width 1*

XamlGrids:
![image](https://github.com/user-attachments/assets/a664b563-9c3c-4370-ab4b-7a80d60172ca)

FuncUIGrids:
![image](https://github.com/user-attachments/assets/8d8e201f-f424-4b29-82c8-5d1d8b2a7f29)

Is there something I'm missing? It doesn't seem like this should be object graph / diff related, but I'm not sure.

**Comment by JaggerJo** [comment]
Hey @marklam 

interesting bug, after looking at it briefly I don't not see anything obvious. Can take a closer look tomorrow

XAML Version:

<img width="436" alt="image" src="https://github.com/user-attachments/assets/6f568f04-ac1d-40e1-b913-958e1389871f" />

FuncUI Version:

<img width="350" alt="image" src="https://github.com/user-attachments/assets/e40aef92-7cab-439f-aa89-3473d1dfe1c3" />

**Comment by marklam** [comment]
Yes, the diagnostics view doesn't give much insight into size scopes, unfortunately.

I pushed a new changeset to make the UI trees match (ItemsControl with ContentPresenter instead of ListBox with ListBoxItem/HostControl) but it doesn't change the sizing.

**Comment by JaggerJo** [comment]
Hey @marklam 

Did you manage to make any progress by yourself?

**Comment by marklam** [comment]
No luck sadly, @JaggerJo 

I had hoed that it was down to the layout hierarchy, but when that didn't fix it, I ended up just picking an adequate-looking fixed size as a workaround.

**Comment by JaggerJo** [comment]
@marklam 

I'm currently pretty time constrained.. I think this needs a full day of attention as it's not obvious why the exact same layout leads to different behaviour. `SharedSizeScope` seems to be not heavily used, I can image it containing some edge cases. 

Would be interesting if the issue still exists if you create the UI tree in code (not FuncUI, raw F#/C#). Then we could isolate if it's XAML magic that's making it, or FuncUI breaking it 🙃

**Comment by edgarfgp** [comment]
Based in what I had to do for Fabulous.Avalonia to support this. I think the logic for Row and Column definitions needs to be [updated](https://github.com/fsprojects/Avalonia.FuncUI/blob/46d44d821093e56da16e114386c4cea02faaf504/src/Avalonia.FuncUI/DSL/Panels/Grid.fs#L4) . 

Context: https://github.com/fabulous-dev/Fabulous.Avalonia/pull/273/files

**Comment by JaggerJo** [comment]
Hey @edgarfgp 👋

 hmm, @marklam also sets SharedSizeGroup columns, see:

https://github.com/marklam/GridSizingScopes/blob/51619b3a6e2ffabedc66985ac8c3c7cb4ea657a6/FuncUIGrids/Program.fs#L42-L47

What am I missing here?
